src/TreeView/useRovingTabIndex.ts

```diff
@@ -150,3 +162,68 @@ export function getLastElement(element: HTMLElement): HTMLElement | undefined {
 
   return last instanceof HTMLElement ? last : undefined
 }
+
+function getNextPageElement(element: HTMLElement): HTMLElement | undefined {
+  const root = element.closest('[role="tree"]')
+  if (!root) {
+    return
+  }
+
+  const items = Array.from(root.querySelectorAll('[role="treeitem"]')).filter((item): item is HTMLElement => {
+    return item.nodeType === Node.ELEMENT_NODE
+  })
+  const scrollContainer = getScrollContainer(root)
+  const visible = items.filter(item => {
+    return isVisible(item.firstElementChild as HTMLElement, scrollContainer)
+  })
+  const pageSize = items.indexOf(visible[visible.length - 1]) - items.indexOf(visible[0]) + 1
+  const page = Math.floor(items.indexOf(element) / pageSize)
+  return items[Math.min(items.length - 1, (page + 1) * pageSize)]
+}
+
+function getPreviousPageElement(element: HTMLElement): HTMLElement | undefined {
+  const root = element.closest('[role="tree"]')
+  if (!root) {
+    return
+  }
+
+  const items = Array.from(root.querySelectorAll('[role="treeitem"]')).filter((item): item is HTMLElement => {
+    return item.nodeType === Node.ELEMENT_NODE
+  })
+  const scrollContainer = getScrollContainer(root)
+  const visible = items.filter(item => {
+    return isVisible(item.firstElementChild as HTMLElement, scrollContainer)
+  })
+  const pageSize = items.indexOf(visible[visible.length - 1]) - items.indexOf(visible[0]) + 1
```

Comment:
We could probably pull out a `getPageSize()` helper function to reduce duplication